src/mongo_odm/nvp.hpp

```diff
@@ -56,16 +74,83 @@
     MONGO_ODM_MAKE_KEYS(Base, MONGO_ODM_NVP(_id), __VA_ARGS__)
 
 #define MONGO_ODM_KEY(value) mongo_odm::hasCallIfFieldIsPresent<decltype(&value), &value>::call()
+#define MONGO_ODM_KEY_BY_VALUE(value) \
+    mongo_odm::hasCallIfFieldIsPresent<decltype(value), value>::call()
+
+#define MONGO_ODM_CHILD1(base, field1) MONGO_ODM_KEY_BY_VALUE(&base::field1)
+
+#define MONGO_ODM_CHILD2(base, field1, field2)                                               \
+    makeNvpWithParent(                                                                       \
+        MONGO_ODM_KEY_BY_VALUE(                                                              \
+            &std::decay_t<typename decltype(MONGO_ODM_CHILD1(base, field1))::type>::field2), \
+        MONGO_ODM_CHILD1(base, field1))
+
+#define MONGO_ODM_CHILD3(base, field1, field2, field3)                                         \
```

Comment:
I'm in agreement.  Do 100 levels check in the generated header and also the check in the script that produces the header.  Probably worth adding a target for the header generation as well in cmake